### ISSUE TITLE
New rule: consider_making_a_field_private (if its not used outside)

### ISSUE THREAD

**Issue by danylo-safonov-solid** [solid-software/solid_lints#93]
This'll help with dead_code

BAD:
```dart
class X {
  // only used internally 
  void x() {
    ...
  }
}
```

GOOD:
```dart
class X {
  // only used internally 
  void _x() {
    ...
  }
}
```

**Comment by kvenn** [comment]
I'm very excited for this one! I could see it being very valuable for any class members. For some reason, Dart doesn't recognize that a member is unused unless it's private. So this would be a double whammy if it enforced all members that _could_ be public were public.

**Comment by illia-romanenko** [comment]
@kevmoo just published a new package that is related to this rule: https://pub.dev/packages/undead. What is proposed here is probably slightly different from this new package, but let's think about how it affects this request and whether depending on undead or other Kevin's work will make better sense. 

I also asked to add a new section of Notable 3rd party libraries to https://lints.solid.software/docs/intro @solid-illiaaihistov - Kevin's work definitely should be at least added there, as it's very relevant to what we are doing here.

**Comment by kevmoo** [comment]
Ooo! @illia-romanenko you gave me an IDEA.

I bet my `undead` code "knows" enough to recommend to make things private. This is a GREAT idea!

**Comment by kevmoo** [comment]
https://github.com/kevmoo/analytica.dart/issues/44
https://github.com/kevmoo/analytica.dart/pull/45
